spark/src/test/java/org/apache/iceberg/spark/data/TestSparkAvroUnions.java

```diff
@@ -141,6 +141,44 @@ public void writeAndValidateOptionalComplexUnion() throws IOException {
 
   @Test
   public void writeAndValidateSingleTypeUnion() throws IOException {
```

Comment:
Could you add another test where the avro schema is a single union of a complex type (e.g. struct/map/array) ?

Reply:
Sure, will add test case for it.